lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py

```diff
@@ -1242,6 +1241,40 @@ def get_service_name_from_init_system(self, service):
             else:
                 return service
 
+    def get_magma_services(self) -> List[str]:
+        """
+        Returns a list of all services managed by magmad and additionally
+        magmad itself, openvswitch-switch and sctpd.
+        """
+        non_magmad_services = [
+            'magmad',
+            'sctpd',
+        ]
+
+        systemd_only_magma_services = [
+            'openvswitch-switch',
+            'magma_dp@envoy',
+        ]
+
+        docker_only_magma_services = [
+            'connectiond',
+            'monitord',
+            'redirectd',
+            'td-agent-bit',
+        ]
+
+        raw_magmad_yml = self.exec_command_output('cat /etc/magma/magmad.yml')
+        magmad_yml = yaml.load(raw_magmad_yml, Loader=yaml.loader.SafeLoader)
+        magma_services = magmad_yml['magma_services'] + non_magmad_services
+
+        if self._init_system == InitMode.SYSTEMD:
+            return magma_services + systemd_only_magma_services
+        elif self._init_system == InitMode.DOCKER:
+            magma_services.remove('dnsd')
+            return magma_services + docker_only_magma_services
+        else:
```

Comment:
nit: is the last else possible? should the following suffice?
```
if self._init_system == InitMode.SYSTEMD:
  return magma_services + systemd_only_magma_services

magma_services.remove('dnsd')
return magma_services + docker_only_magma_services
```

Reply:
No, this is more a fail save. But this should not really be possible, you're right.